examples/700-go-hook/modules/001-module-go-hooks/hooks/go_hooks.go

```diff
@@ -1,183 +1,90 @@
-				//JqFilter:             ".spec",
-				FilterFunc: func(obj *unstructured.Unstructured) (string, error) {
-					var pod v1.Pod
-					err := runtime.DefaultUnstructuredConverter.
-						FromUnstructured(obj.UnstructuredContent(), &pod)
-					if err != nil {
-						return "", err
-					}
-
-					res, err := json.Marshal(pod.Spec)
-					if err != nil {
-						return "", err
-					}
-					fmt.Printf("got pod.spec: %s\n", string(res))
-					return string(res), nil
-				},
-				ExecuteHookOnEvents:          []types.WatchEventType{types.WatchEventAdded, types.WatchEventModified, types.WatchEventDeleted},
-				ExecuteHookOnSynchronization: true,
-				Handler: func(input *sdk.BindingInput) (*sdk.BindingOutput, error) {
-					input.LogEntry.Infof("Hello from on_kube.pods2! I have %d snapshots for '%s' event\n",
-						len(input.BindingContext.Snapshots),
-						input.BindingContext.WatchEvent)
-
-					vs := values_store.NewValuesStoreFromValues(input.Values)
-
-					input.LogEntry.Infof("go_hooks kube hook got values: %s", vs.GetAsYaml())
-
-					return nil, nil
-				},
+				Name:                         "pods",
+				ApiVersion:                   "v1",
+				Kind:                         "Pods",
+				Filterable:                   &podSpecFilteredObj{},
+				ExecuteHookOnSynchronization: go_hook.Bool(true),
 			},
 		},
 
-		Schedule: []sdk.ScheduleConfig{
+		Schedule: []go_hook.ScheduleConfig{
 			{
 				Name:    "metrics",
 				Crontab: "*/5 * * * * *",
-				Handler: h.SendMetrics,
 			},
 		},
-	})
-}
-
-// Go hook has no kubectl, but it can initialize its own kubernetes client!
-func (h *GoHook) initKubeClient() error {
-	if h.kubeClient != nil {
-		return nil
 	}
-
-	h.kubeClient = kube.NewKubernetesClient()
-	h.kubeClient.WithContextName(app.KubeContext)
-	h.kubeClient.WithConfigPath(app.KubeConfig)
-	h.kubeClient.WithRateLimiterSettings(app.KubeClientQps, app.KubeClientBurst)
-	// Initialize kube client for kube events hooks.
-	err := h.kubeClient.Init()
-	if err != nil {
-		return err
-	}
-	return nil
 }
 
-func (h *GoHook) SendMetrics(input *sdk.BindingInput) (*sdk.BindingOutput, error) {
-	err := h.initKubeClient()
-	if err != nil {
-		input.LogEntry.Errorf("Fatal: initialize kube client: %s", err)
-		return nil, err
-	}
-
-	podList, err := h.kubeClient.CoreV1().Pods("").List(metav1.ListOptions{})
-	if err != nil {
-		input.LogEntry.Errorf("Fatal: cannot list pods: %s", err)
-		return nil, err
+func (h *GoHook) Metadata() *go_hook.HookMetadata {
+	return &go_hook.HookMetadata{
+		Name:       "go_hook.go",
+		Path:       "001-module-go-hooks/hooks/go_hook.go",
+		Module:     true,
+		ModuleName: "module-go-hooks",
 	}
+}
 
-	input.LogEntry.Infof("Get %d pods:", len(podList.Items))
-	for i, pod := range podList.Items {
-		input.LogEntry.Infof("%02d. Pod/%s in ns/%s", i, pod.Name, pod.Namespace)
+func (h *GoHook) Run(input *go_hook.HookInput) error {
+	for _, o := range input.Snapshots["pods"] {
+		podSpec := o.FilterResult.(*podSpecFilteredObj)
```

Comment:
Will it work? o.FilterResult is a *v1.PodSpec.

Reply:
It will. Why shouldn't it?